core/domain/user_services.py

```diff
@@ -656,3 +656,38 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+def get_weekly_dashboard_stats(user_id):
```

Comment:
That last paragraph isn't guaranteed, though. We might stop the job for some reason, or the job may fail, or something, so we can't really assume. That's why we have the explicit keys, so each entry basically represents a data point.